File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/BorderFlashEffect.ts
import { FairyGUI } from "csharp";
import Color from "../../DataDefine/Color";
import { View_BorderFlashEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";

export class BorderFlashEffect extends AbstractEffect{
    public Loader:FairyGUI.GComponent
    public constructor(private _Color:Color,private _Additive?:boolean) {
        super();
        this.Loader = <FairyGUI.GComponent>View_BorderFlashEffect.CreateInstance();
        this.Loader.GetTransition("t0").Play();
        let image = this.Loader.GetChild("n0").asImage;
        image.color = _Color.UnityColor;
        if(_Additive == true)
            image.blendMode = FairyGUI.BlendMode.Add;
    }
    Update() {
        super.Update();
        if(this.Duration >= 1) {
            this.IsDone = true;
            this.Loader.Dispose();
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/BorderFlashEffect.cs
using FairyGUI;
using ModTheSpire.ModTheSpire_Effect;
using UnityEngine;

namespace ModTheSpire.Effect.Combat
{
    public class BorderFlashEffect : AbstractEffect
    {
        private Color mColor;
        private bool mAdditive;
        private GComponent mLoader;
        public BorderFlashEffect(Color color, bool additive = false)
        {
            mColor = color;
            mAdditive = additive;
            mLoader = View_BorderFlashEffect.CreateInstance();
            mLoader.GetTransition("t0").Play();
            var image = mLoader.GetChild("n0").asImage;
            image.color = mColor;
            if(mAdditive == true)
                image.blendMode = BlendMode.Add;
        }

        public override void Update()
        {
            base.Update();
            if(this.Duration >= 1)
                Finish();
        }

        public override void Dispose()
        {
            base.Dispose();
            mLoader.Dispose();
        }
    }
}